src/rgw/rgw_cr_rados.cc

```diff
@@ -491,6 +492,51 @@ int RGWAsyncGetBucketInstanceInfo::_send_request()
   return 0;
 }
 
+static std::string normalize_shard_marker(const std::string& marker)
+{
+  // markers may be formatted with '<shard-id>#' at the beginning.
+  // CLSRGWIssueBILogTrim would fix this for us, but it's synchronous
+  auto p = marker.find(BucketIndexShardsManager::KEY_VALUE_SEPARATOR);
+  if (p == marker.npos) {
+    return marker;
+  }
+  return marker.substr(p + 1);
+}
```

Comment:
Surely we defined a similar method elsewhere? if not, then the place for it is wherever BucketIndexShardsManager is defined I think